docs/standard/serialization/binaryformatter-migration-guide/wpf-applications.md

```diff
@@ -0,0 +1,104 @@
+ms.date: 8/05/2024
+no-loc: [BinaryFormatter, WPF]
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "WPF"
+---
+
+# WPF Migration Guide – Binary Formatter 
+
+ 
+## Binary Formatter Removal 
+
+In .NET 9.0, the entire `Binaryformatter` infrastructure will be removed from the product and moved as a separate Nuget  package. For more information about the risks `Binaryformatter` poses and our decision on its removal, see [BinaryFormatter migration guide](index.md). 
+
+ 
+## How Binary Formatter affects WPF 
+
+WPF uses `Binaryformatter` to serialize/deserialize the data for scenarios such clipboard, drag/drop and store data for Avalon Binding in Journal.  
+
+WPF has and is still currently using `Binaryformatter` code internally for all the above scenarios, however we have taken some measures to mitigate the usage of `Binaryformatter`.  
+
+We have added an implementation based on WinForms to handle common primitive types, including primitive arrays and lists. Usage of these types in the mentioned scenarios will continue to work without using `Binaryformatter`, similarly as before. 
+
+Primitive types include: 
+- `bool` 
+- `byte` 
+- `char` 
+- `decimal` 
+- `double` 
+- `int` 
+- `sbyte` 
+- `float` 
+- `TimeSpan` 
+- `DateTime` 
+- `uint` 
+- `string` 
+- `nint` 
+- `nuint` 
+- `long`
+- `ulong`
+- `short`
+- `ushort` 
+
+Additional types include: 
+- `PointF` 
+- `RectangleF`  
+  
+## Where Binary Formatter is used in WPF 
+
+### Clipboard  
+
+In WPF application if you are using a clipboard to copy or paste the data into the application or Set/Get data from a DataObject or using drag/drop. 
+
+If the data format ([DataFromats](https://learn.microsoft.com/en-us/dotnet/api/system.windows.dataformats?view=windowsdesktop-8.0)) is part of the following list, it will not require BinaryFormatter to handle: 
+- `Stream` 
+- `Text`, `Rtf`, `OemText`, `CommaSeparatedValue` 
+- `UnicodeText` 
+- `FileDrop` 
+- `FileName` 
+- `BitmapSource` 
+
+Apart from these formats, if the type is serializable and not included in the above list of types handled by the new implementation, `Binaryformatter` will be used for handling. 
+
+In scenarios where `Binaryformatter` is utilized, its removal will result in developers encountering an exception indicating that `Binaryformatter` has been removed. 
+
+You can refer to the function where we have used `Binaryformatter` as fallback to read/save object to handle:  [SaveObjectToHandle](https://github.com/dotnet/wpf/blob/0354a597996adae43b12efc72bd705f76d4ba497/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs#L1677) and  [ReadObjectFromHandle](https://github.com/dotnet/wpf/blob/0354a597996adae43b12efc72bd705f76d4ba497/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs#L3051) 
+
+Additionally, if you encounter an exception stating, 'Data stored in clipboard is invalid,' it may be because `Binaryformatter` was used to save the data. 
```

Comment:
```suggestion
Additionally, if you encounter an exception stating, 'Data stored in clipboard is invalid,' it may be because `Binaryformatter` was used to save the data. 
```
```suggestion
Additionally, if you encounter an exception that states "Data stored in clipboard is invalid," it might be because `BinaryFormatter` was used to save the data. 
```